scripts/core-ng-module-generator.js

```diff
@@ -0,0 +1,172 @@
+    const templateRender = Mustache.render(template, templateData);
+    fs.writeFileSync(filePath, templateRender);
+  }
+}
+
+function createRootIndex(moduleData) {
+  if (moduleData) {
+    const templateData = {
+      modules: Object.keys(moduleData).map(md => ({ moduleDirectoryName: md })),
+    };
+
+    createFileOnTemplate(rootIndexTemplate, templateData, 'index.ts', TARGET_PATH);
+  }
+}
+
+function createIndexes(moduleData) {
+  if (moduleData) {
+    Object.keys(moduleData).forEach(moduleDirectory => {
+      const templateData = {
+        directives: Object.keys(moduleData[moduleDirectory]).map(md => ({
+          directiveFileName: md,
+        })),
+        moduleFileName: 'cds-' + moduleDirectory,
+      };
+
+      createFileOnTemplate(indexTemplate, templateData, 'index.ts', TARGET_PATH, moduleDirectory);
+    });
+
+    createRootIndex(moduleData);
+  }
+}
+
+function createModules(moduleData) {
+  if (moduleData) {
+    const rootModuleTemplateData = { modules: [] };
+    Object.keys(moduleData).forEach(moduleDirectory => {
+      const templateData = {
+        directives: Object.keys(moduleData[moduleDirectory]).map(md => ({
+          directiveFileName: md,
+          directiveClassName: moduleData[moduleDirectory][md],
+        })),
+        moduleClassName: getDirectiveClassName(moduleDirectory, 'Cds', 'Module'),
+      };
+
+      rootModuleTemplateData.modules.push({ moduleClassName: templateData.moduleClassName, moduleDirectory });
+
+      const moduleFileName = 'cds-' + moduleDirectory + '.module.ts';
+      createFileOnTemplate(moduleTemplate, templateData, moduleFileName, TARGET_PATH, moduleDirectory);
+    });
+
+    createFileOnTemplate(rootModuleTemplate, rootModuleTemplateData, 'cds.module.ts', TARGET_PATH);
+
+    return moduleData;
+  }
+}
+
+function createDirectives(componentDirectories) {
+  if (componentDirectories) {
+    const customElementsData = JSON.parse(fs.readFileSync(SOURCE_PATH));
+
+    return customElementsData.tags.reduce((moduleData, tag) => {
+      const templateData = {
+        tagName: tag.name,
+        directiveClassName: getDirectiveClassName(tag.name, '', 'Directive'),
+        props: tag.properties,
+        hasProps: !!tag.properties,
+      };
+
+      const directory = getDiretoryName(tag.path);
+
+      createFileOnTemplate(directiveTemplate, templateData, getDirectiveName(tag.name), TARGET_PATH, directory);
+
+      if (componentDirectories.includes(directory)) {
+        if (moduleData[directory]) {
+          moduleData[directory][templateData.tagName] = [templateData.directiveClassName];
+        } else {
+          moduleData[directory] = { [templateData.tagName]: templateData.directiveClassName };
+        }
+      }
+
+      return moduleData;
+    }, {});
+  }
+}
+
+function createComponentDirectories() {
+  if (fs.existsSync(SOURCE_PATH)) {
+    const customElementsData = JSON.parse(fs.readFileSync(SOURCE_PATH));
```

Comment:
same here `JSON.parse` could throw `SyntaxError` and kill the `core-ng-module-generator` task

Reply:
Wouldnt we want that to throw and kill the task so we know something is wrong? Or catch and throw a specific exception?